src/behave/behaveTestRunner.ts

```diff
@@ -0,0 +1,209 @@
+import * as path from 'path';
+
+import {
+    TestEvent, TestSuiteInfo
+} from 'vscode-test-adapter-api';
+
+import { ArgumentParser } from 'argparse';
+import { IWorkspaceConfiguration } from '../configuration/workspaceConfiguration';
+import { IEnvironmentVariables, EnvironmentVariablesLoader } from '../environmentVariablesLoader';
+import { ILogger } from '../logging/logger';
+import { IProcessExecution, runProcess } from '../processRunner';
+import { IDebugConfiguration, ITestRunner } from '../testRunner';
+import { empty } from '../utilities/collections';
+import { setDescriptionForEqualLabels } from '../utilities/tests';
+import { parseTestStates } from './behaveTestJsonParser';
+import { parseTestSuites } from './behaveTestJsonParser';
+import { runModule } from '../pythonRunner';
+
+// --- Behave Exit Codes ---
+// 0: All tests were collected and passed successfully
+// 1: Some tests have failed
+const BEHAVE_NON_ERROR_EXIT_CODES = [0, 1];
+
+const DISCOVERY_OUTPUT_PLUGIN_INFO = {
```

Comment:
Not sure what you mean. Behave exit codes are not documented (to my current knowledge). I have seen 0 and 1 as valid exit codes and unfortunately 0 when an error occurs (that then spits out an error string instead of json data).

Reply:
I meant DISCOVERY_OUTPUT_PLUGIN_INFO constant 🙂

Reply:
It is used in the loadEnvironmentVariables in the same file (line 154 and 159).

Reply:
Yes, it's referenced, but it makes sense only for pytest - pytest is run with a custom plugin (in resources/python folder). Since behave does not need any custom plugins this constant and its usages are not actually useful. If you have any doubts, I can suggest writing tests first and then try removing this, I'm pretty sure tests won't break.